packages/react/src/internal/usePreviousValue.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright IBM Corp. 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useEffect, useRef } from 'react';
+
+/**
+ * Stores the previous value of a given input.
+ *
+ * @param value - The current value.
+ * @returns The value before the current render.
+ *
+ * @example
+ * const [count, setCount] = useState(0);
+ * const prevCount = usePreviousValue(count);
+ */
+export const usePreviousValue = <T>(value: T): T | undefined => {
+  const ref = useRef<T | undefined>(undefined);
+
+  useEffect(() => {
+    ref.current = value;
+  }, [value]);
+
+  return ref.current;
+};
```

Comment:
If this util is going to be added, do tests need to be written for it?

Reply:
Hmm, you're making me regret copying over this function as opposed to using a library like https://usehooks.com/useprevious.  But I added a unit test.

Reply:
Apologies. If https://github.com/uidotdev/usehooks/blob/90fbbb4cc085e74e50c36a62a5759a40c62bb98e/index.js#L1017-L1027 is the code for that hook, I would be hesitant to use it. It doesn't follow the rules of hooks.

Reply:
I guess it technically breaks the rules of hooks, but 

```js
  if (value !== current) {
    setPrevious(current);
    setCurrent(value);
  }
```

is essentially equivalent to

```js
useEffect(() => {
  if (value !== current) {
    setPrevious(current);
    setCurrent(value);
  }
}, [current, value]);
```

The more interesting thing to me, assuming I'm reading this correctly, is that the behavior is different.  

`open` and `usePreviousValue(open)` are only different for a single render, after which `usePreviousValue(open)` catches up to the current value.

Whereas `open` and `usePrevious(open)` are always different, no matter how many times the component renders.